sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TransmissionStateManager.cs

```diff
@@ -0,0 +1,94 @@
+
+        private DateTimeOffset _nextMinTimeToUpdateConsecutiveErrors = DateTimeOffset.MinValue;
+
+        private readonly System.Timers.Timer _backOffIntervalTimer;
+
+        private double _syncBackOffIntervalCalculation;
+
+        internal TransmissionState State { get; private set; }
+
+        private TransmissionStateManager()
+        {
+            _backOffIntervalTimer = new();
+            _backOffIntervalTimer.Elapsed += RestartTransmission;
+            _backOffIntervalTimer.AutoReset = false;
+            State = TransmissionState.Closed;
+        }
+
+        /// <summary>
+        /// Prevents transmitting data to backend.
+        /// </summary>
+        private void OpenTransmission()
+        {
+            State = TransmissionState.Open;
+        }
+
+        internal void RestartTransmission(Object source, System.Timers.ElapsedEventArgs e)
+        {
+            _consecutiveErrors = 0;
+
+            // Reset _sync so that the threads can open the transmission again if needed.
+            _syncBackOffIntervalCalculation = 0;
+            State = TransmissionState.Closed;
+        }
+
+        internal void EnableBackOff(Response? response)
+        {
+            if (Interlocked.Exchange(ref _syncBackOffIntervalCalculation, 1) == 0)
+            {
+                // Do not increase number of errors more often than minimum interval (MinDelayInMilliseconds).
+                // since we can have 4 parallel transmissions (logs, metrics, traces and offline storage tranmission) and all of them most likely would fail if we have intermittent error.
+                if (DateTimeOffset.UtcNow > _nextMinTimeToUpdateConsecutiveErrors)
+                {
+                    _consecutiveErrors++;
+                    _nextMinTimeToUpdateConsecutiveErrors = DateTimeOffset.UtcNow + _minIntervalToUpdateConsecutiveErrors;
+
+                    // If backend responded with a retryAfter header we will use it
+                    // else we will calculate by increasing time interval exponentially.
+                    var retryAfterInterval = HttpPipelineHelper.GetRetryIntervalTimespan(response);
+                    var backOffTimeInterval = retryAfterInterval != TimeSpan.MinValue ? retryAfterInterval : GetBackOffTimeInterval();
+
+                    OpenTransmission();
+
+                    _backOffIntervalTimer.Interval = backOffTimeInterval.TotalMilliseconds;
+
+                    _backOffIntervalTimer.Start();
+                }
+
+                Interlocked.Exchange(ref _syncBackOffIntervalCalculation, 0);
+            }
+        }
+
+        internal TimeSpan GetBackOffTimeInterval()
+        {
+            double delayInMilliseconds = MinDelayInMilliseconds;
+            if (_consecutiveErrors > 1)
+            {
+                double backOffSlot = (Math.Pow(2, _consecutiveErrors) - 1) / 2;
+                var backOffDelay = _random.Next(1, (int)Math.Min(backOffSlot * MinDelayInMilliseconds, int.MaxValue));
```

Comment:
Why are we picking a random value?

Reply:
I guess it's for traffic shaping (e.g. several w3wp.exe instances started at almost the same time).

Reply:
https://en.wikipedia.org/wiki/Exponential_backoff#:~:text=etc.%5B2%5D-,Collision%20avoidance,-%5Bedit%5D
In our case this^ can happen when we have many number of application instances sending data to same endpoint.